stdlib/public/core/StringUTF16.swift

```diff
@@ -45,24 +47,28 @@ extension String {
     // TODO: swift-3-indexing-model - add docs
     @warn_unused_result
     public func next(i: Index) -> Index {
+      // FIXME: swift-3-indexing-model: range check i?
       return Index(_offset: i._offset.advanced(by: 1))
```

Comment:
Yeah I wasn't thinking far enough that the Index was actually an Int... will update.